server/src/controllers/user/login.ts

```diff
@@ -1,5 +1,34 @@
-// import { signInSchema } from '../../utils';
+import { Request, Response, NextFunction } from 'express';
+import { compare } from 'bcrypt';
+import { checkEmailQuery } from '../../database/ quieres';
+import { signInSchema, signToken } from '../../utils';
+import { customError } from '../errors';
 
-// const login = (req, res) =>{
+const login = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const { email, password } = await signInSchema.validateAsync(req.body, { abortEarly: false });
```

Comment:
you can add option object {abort early} to the schema file, i think it is better than add it here.